### PR TITLE
feat: add echo for springboot http template

### PR DESCRIPTION
**Pull request by mcruzdev**
# Changes
- :gift: Add echo response to template `springboot` `http`

```release-note
springboot http template generates echo, not uppercase function
```

### CODE REVIEW

**Comment by mcruzdev** [comment]
> @mcruzdev this commit contains idea files in templates (in the zz_* file) you need to remove those files are regenerate the zz_* file.

thank you @matejvasek, done!

**Comment by matejvasek** [comment]
@mcruzdev there are still files that shouldn't be there https://github.com/knative-sandbox/kn-plugin-func/runs/6937029324?check_suite_focus=true

**Comment by matejvasek** [comment]
@mcruzdev  https://github.com/knative-sandbox/kn-plugin-func/blob/main/docs/CONTRIBUTING.md#templates

File: schema/func_yaml-schema.json [hunk 1]
```diff
@@ -1,321 +0,0 @@
-{
```
Comment (matejvasek) [inline schema/func_yaml-schema.json:1]: @mcruzdev I don't think this should be deleted.

**Comment by matejvasek** [comment]
@mcruzdev there are some lint style issues with trialing white-spaces, after fixing that is should be all right I think.

**Comment by matejvasek** [comment]
there are .idea/ files again https://github.com/knative-sandbox/kn-plugin-func/runs/6938140623?check_suite_focus=true

**Comment by matejvasek** [comment]
you need also to update e2e test https://github.com/knative-sandbox/kn-plugin-func/blob/ccfa02afdff745c240d4d5b5580e41c7c2cca691/test/_e2e/trigger_http_test.go

**Comment by matejvasek** [comment]
@mcruzdev you need to fix the e2e test, then it will be ok hopefully :crossed_fingers:

File: test/_e2e/trigger_http_test.go [hunk 2]
```diff
@@ -86,7 +86,7 @@ var defaultFunctionsHttpValidators = []FunctionHttpResponsivenessValidator{
 		expects:   `{"message":"hello"}`,
 	},
 	{runtime: "springboot",
-		targetUrl: "%s/uppercase",
+		targetUrl: "%s/echo",
 		method:    "POST",
 		bodyData:  "message=hello",
 		expects:   `MESSAGE=HELLO`,
```
Comment (matejvasek) [inline test/_e2e/trigger_http_test.go:92]: Shouldn't be this lowercase?

Comment (mcruzdev) [reply test/_e2e/trigger_http_test.go:92]: Thank you for your help @matejvasek.

I changed the `/uppercase` endpoint to `/echo`.

Comment (mcruzdev) [reply test/_e2e/trigger_http_test.go:92]: I got it, e2e_tests looks for https://github.com/mcruzdev/kn-plugin-func/blob/mcruzdev-1062-echo/test/_e2e/update_templates/springboot/http/src/main/java/functions/CloudFunctionApplication.java file.

Comment (mcruzdev) [reply test/_e2e/trigger_http_test.go:92]: I will change the expected result, thank you.

Comment (matejvasek) [reply test/_e2e/trigger_http_test.go:92]: expects:   `MESSAGE=HELLO`,
should be:
		expects:   `message=hello`,

Comment (matejvasek) [reply test/_e2e/trigger_http_test.go:92]: > I got it, e2e_tests looks for https://github.com/mcruzdev/kn-plugin-func/blob/mcruzdev-1062-echo/test/_e2e/update_templates/springboot/http/src/main/java/functions/CloudFunctionApplication.java file.

thats cloudevents thing not plain http thing

File: templates/springboot/http/src/main/java/functions/CloudFunctionApplication.java [hunk 3]
```diff
@@ -13,8 +15,22 @@ public static void main(String[] args) {
   }
 
   @Bean
-  public Function<String, String> uppercase() {
-    return String::toUpperCase;
-  }
+  public Function<Message<String>, String> echo() {
+    return (inputMessage) -> {
 
+      var stringBuilder = new StringBuilder();
```
Comment (matejvasek) [inline templates/springboot/http/src/main/java/functions/CloudFunctionApplication.java:21]: this does a lot, shouldn't we just echo body

Comment (matejvasek) [reply templates/springboot/http/src/main/java/functions/CloudFunctionApplication.java:21]: @mcruzdev  ^^^

Comment (matejvasek) [reply templates/springboot/http/src/main/java/functions/CloudFunctionApplication.java:21]: otherwise you must fix expected in e2e test

Comment (matejvasek) [reply templates/springboot/http/src/main/java/functions/CloudFunctionApplication.java:21]: or maybe not if there are no headers...

Comment (matejvasek) [reply templates/springboot/http/src/main/java/functions/CloudFunctionApplication.java:21]: the test is passing

Comment (matejvasek) [reply templates/springboot/http/src/main/java/functions/CloudFunctionApplication.java:21]: it's ok I guess

Comment (mcruzdev) [reply templates/springboot/http/src/main/java/functions/CloudFunctionApplication.java:21]: It is the same thing that `go:http` does.

Comment (mcruzdev) [reply templates/springboot/http/src/main/java/functions/CloudFunctionApplication.java:21]: > 

I would like to add `echo:  message=hello` for expects

**Comment by matejvasek** [comment]
@mcruzdev thank you for your contribution!

**Comment by mcruzdev** [comment]
> @mcruzdev thank you for your contribution!

Thank you for your patience and help!

**Pull request merged**